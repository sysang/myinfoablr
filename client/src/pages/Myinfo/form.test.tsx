import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

// We're using our own custom render function and not RTL's render.
import { renderWithProviders } from '../../test-utils'

import MyinfoForm, {
    ContactInfo,
    PersonalInfo,
    IncomeInfo,
} from './form'


describe('ContactInfo', () => {
    test('renders ContactInfo component', () => {
        const myinfo = {
            "mobileno": "97399245",
            "email": "myinfotesting@gmail.com",
            "regadd_block": "102",
            "regadd_street": "BEDOK NORTH AVENUE 4",
            "regadd_building": "PEARL GARDEN",
            "regadd_unit": "",
            "regadd_postal": "",
            "regadd_type": "",
        }
        const readOnly = {
            "mobileno": true,
            "email": true,
            "regadd_block": true,
            "regadd_street": true,
            "regadd_building": true,
            "regadd_unit": false,
            "regadd_postal": false,
            "regadd_type": false,
        }

        const tree = render(<ContactInfo readOnly={readOnly} myinfo={myinfo}/>);

        expect(screen.getByTestId('mobileno')).toHaveValue('97399245')
        expect(screen.getByTestId('email')).toHaveValue('myinfotesting@gmail.com')
        expect(screen.getByTestId('regadd_block')).toHaveValue('102')
        expect(screen.getByTestId('regadd_building')).toHaveValue('PEARL GARDEN')
        expect(screen.getByTestId('regadd_unit')).toHaveValue('')
        expect(screen.getByTestId('regadd_postal')).toHaveValue('')
        expect(screen.getByTestId('regadd_type')).toHaveValue('')

        expect(tree).toMatchSnapshot();
    });
});


describe('PersonalInfo', () => {
    test('renders PersonalInfo component', () => {
        const myinfo = {
            "uinfin": "S9812381D",
            "name": "TAN XIAO HUI",
            "sex": "FEMALE",
            "dob": "1998-06-06",
            "birthcountry": "SINGAPORE",
            "residentialstatus": "CITIZEN",
            "nationality": "SINGAPORE CITIZEN",
            "race": "CHINESE",
        }
        const readOnly = {
            "uinfin": true,
            "name": true,
            "sex": true,
            "dob": true,
            "birthcountry": true,
            "residentialstatus": true,
            "nationality": true,
            "race": true,
        }

        const tree = render(<PersonalInfo readOnly={readOnly} myinfo={myinfo}/>);

        expect(screen.getByTestId('uinfin')).toHaveValue('S9812381D')
        expect(screen.getByTestId('name')).toHaveValue('TAN XIAO HUI')
        expect(screen.getByTestId('sex')).toHaveValue('FEMALE')
        expect(screen.getByTestId('dob')).toHaveValue('1998-06-06')
        expect(screen.getByTestId('birthcountry')).toHaveValue('SINGAPORE')
        expect(screen.getByTestId('residentialstatus')).toHaveValue('CITIZEN')
        expect(screen.getByTestId('nationality')).toHaveValue('SINGAPORE CITIZEN')
        expect(screen.getByTestId('race')).toHaveValue('CHINESE')

        expect(tree).toMatchSnapshot();
    });
});


describe('IncomeInfo', () => {
    test('renders IncomeInfo component', () => {
        const myinfo = {
            "noas": [
                {
                    "yearofassessment": "2021",
                    "employment": 64400,
                    "trade": 0,
                    "interest": 0,
                    "rent": 0,
                    "amount": 64400,
                    "taxclearance": "N"
                },
                {
                    "yearofassessment": "2020",
                    "employment": 36112.8,
                    "trade": 0,
                    "interest": 0,
                    "rent": 0,
                    "amount": 53700,
                    "taxclearance": "N"
                }
            ],
            "ownerprivate": false,
            "cpfbalances_oa": 58839.75,
            "cpfbalances_sa": 15349.5,
            "cpfbalances_ma": 20466,
            "cpfcontributions": [
                {
                    "month": "2021-08",
                    "date": "2021-08-25",
                    "amount": 1425,
                    "employer": "CRYSTAL HORSE INVEST PTE LTD"
                },
                {
                    "month": "2021-09",
                    "date": "2021-09-25",
                    "amount": 1425,
                    "employer": "CRYSTAL HORSE INVEST PTE LTD"
                }
            ]
        }
        const readOnly = {}

        const tree = render(<IncomeInfo readOnly={readOnly} myinfo={myinfo}/>);

        const naosSectionElm = screen.getByTestId('noas')

        expect(naosSectionElm).toHaveTextContent('2021')
        expect(naosSectionElm).toHaveTextContent('64400')
        expect(naosSectionElm).toHaveTextContent('36112.8')
        expect(naosSectionElm).toHaveTextContent('53700')

        expect(screen.getByTestId('cpfbalances_oa')).toHaveValue('58839.75')
        expect(screen.getByTestId('cpfbalances_sa')).toHaveValue('15349.5')
        expect(screen.getByTestId('cpfbalances_ma')).toHaveValue('20466')

        const cpfcontributionsElm = screen.getByTestId('cpfcontributions')

        expect(cpfcontributionsElm).toHaveTextContent('Aug 2021')
        expect(cpfcontributionsElm).toHaveTextContent('25 Aug 2021')
        expect(cpfcontributionsElm).toHaveTextContent('1425')
        expect(cpfcontributionsElm).toHaveTextContent('CRYSTAL HORSE INVEST PTE LTD')

        expect(tree).toMatchSnapshot();
    });
});

describe('MyinfoForm', () => {
    test('It should update myinfo data', () => {
        const readOnly = {
            "mobileno": false,
            "email": false,
            "regadd_block": false,
            "regadd_street": false,
            "regadd_building": false,
            "regadd_unit": false,
            "regadd_postal": false,
            "regadd_type": false,
        }

        const tree = renderWithProviders(
            <MyinfoForm readOnly={readOnly} />)

        let inputElm
        const inputVal = '123456789'

        for (let field in readOnly){
            inputElm = screen.getByTestId('email')
            fireEvent.change(inputElm, {
                target: { value: inputVal},
            });
            expect(inputElm).toHaveValue(inputVal)
        }
    })
})
