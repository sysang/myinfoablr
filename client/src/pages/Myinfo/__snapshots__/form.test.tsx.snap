// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo renders ContactInfo component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h5
          class="my-4"
        >
          Contact Info
        </h5>
        <label
          class="text-black-50 mb-1 form-label"
          for="mobileno"
        >
          Mobile Number
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="mobileno"
            id="mobileno"
            readonly=""
            type="tel"
            value="97399245"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="email"
        >
          Email Address
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="email"
            id="email"
            readonly=""
            type="email"
            value="myinfotesting@gmail.com"
          />
        </div>
        <h5
          class="my-4"
        >
          Registered Address
        </h5>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_block"
        >
          Block Number
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="regadd_block"
            id="regadd_block"
            readonly=""
            type="text"
            value="102"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_street"
        >
          Street Name
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="regadd_street"
            id="regadd_street"
            readonly=""
            type="text"
            value="BEDOK NORTH AVENUE 4"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_building"
        >
          Building Name
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="regadd_building"
            id="regadd_building"
            readonly=""
            type="text"
            value="PEARL GARDEN"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_unit"
        >
          Floor & Unit No
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control"
            data-testid="regadd_unit"
            id="regadd_unit"
            type="text"
            value=""
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_postal"
        >
          Postal Code
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control"
            data-testid="regadd_postal"
            id="regadd_postal"
            type="text"
            value=""
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="regadd_type"
        >
          Type of Housing
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control"
            data-testid="regadd_type"
            id="regadd_type"
            type="text"
            value=""
          />
        </div>
        <div>
          <button
            class="button text-uppercase btn btn-primary"
            type="button"
          >
            Continue
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h5
        class="my-4"
      >
        Contact Info
      </h5>
      <label
        class="text-black-50 mb-1 form-label"
        for="mobileno"
      >
        Mobile Number
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="mobileno"
          id="mobileno"
          readonly=""
          type="tel"
          value="97399245"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="email"
      >
        Email Address
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="email"
          id="email"
          readonly=""
          type="email"
          value="myinfotesting@gmail.com"
        />
      </div>
      <h5
        class="my-4"
      >
        Registered Address
      </h5>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_block"
      >
        Block Number
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="regadd_block"
          id="regadd_block"
          readonly=""
          type="text"
          value="102"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_street"
      >
        Street Name
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="regadd_street"
          id="regadd_street"
          readonly=""
          type="text"
          value="BEDOK NORTH AVENUE 4"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_building"
      >
        Building Name
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="regadd_building"
          id="regadd_building"
          readonly=""
          type="text"
          value="PEARL GARDEN"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_unit"
      >
        Floor & Unit No
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control"
          data-testid="regadd_unit"
          id="regadd_unit"
          type="text"
          value=""
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_postal"
      >
        Postal Code
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control"
          data-testid="regadd_postal"
          id="regadd_postal"
          type="text"
          value=""
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="regadd_type"
      >
        Type of Housing
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control"
          data-testid="regadd_type"
          id="regadd_type"
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          class="button text-uppercase btn btn-primary"
          type="button"
        >
          Continue
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`IncomeInfo renders IncomeInfo component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h5
          class="my-4"
        >
          Notice of Assessment (History)
        </h5>
        <div
          data-testid="noas"
        >
          <div
            class="mb-3 row"
          >
            <div
              class="text-start fw-bolder col-md-6"
            >
              Year of Assessment
            </div>
            <div
              class="text-end col-md-3"
            >
              2021
            </div>
            <div
              class="text-end col-md-3"
            >
              2020
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start col-md-6"
            >
              Employment
            </div>
            <div
              class="text-end col-md-3"
            >
              64400
            </div>
            <div
              class="text-end col-md-3"
            >
              36112.8
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start col-md-6"
            >
              Trade
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start col-md-6"
            >
              Interest
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start col-md-6"
            >
              Rent
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
            <div
              class="text-end col-md-3"
            >
              0
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start fw-bolder col-md-6"
            >
              Total income
            </div>
            <div
              class="text-end col-md-3"
            >
              64400
            </div>
            <div
              class="text-end col-md-3"
            >
              53700
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="text-start fw-bolder col-md-6"
            >
              Tax Clearance
            </div>
            <div
              class="text-end col-md-3"
            >
              N
            </div>
            <div
              class="text-end col-md-3"
            >
              N
            </div>
          </div>
        </div>
        <h5
          class="my-4"
        >
          Other income Information
        </h5>
        <label
          class="text-black-50 mb-1 form-label"
          for="ownerprivate"
        >
          Ownership of Private Residential Property
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="ownerprivate"
            id="ownerprivate"
            readonly=""
            type="text"
            value="No"
          />
        </div>
        <h5
          class="my-4"
        >
          CPF Account Balance
        </h5>
        <div
          class="mb-3 input-group"
        >
          <span
            class="input-group-text"
            id="basic-addon1"
          >
            Ordinary Account (OA)(S$)
          </span>
          <input
            aria-describedby="basic-addon1"
            class="text-end form-control"
            data-testid="cpfbalances_oa"
            readonly=""
            type="text"
            value="58839.75"
          />
        </div>
        <div
          class="mb-3 input-group"
        >
          <span
            class="input-group-text"
            id="basic-addon2"
          >
            Special Account (SA)(S$)
          </span>
          <input
            aria-describedby="basic-addon2"
            class="text-end form-control"
            data-testid="cpfbalances_sa"
            readonly=""
            type="text"
            value="15349.5"
          />
        </div>
        <div
          class="mb-3 input-group"
        >
          <span
            class="input-group-text"
            id="basic-addon3"
          >
            Medisave Account (MA)(S$)
          </span>
          <input
            aria-describedby="basic-addon3"
            class="text-end form-control"
            data-testid="cpfbalances_ma"
            readonly=""
            type="text"
            value="20466"
          />
        </div>
        <h5
          class="my-4"
        >
          CPF Contributions History
        </h5>
        <table
          class="cpf_talbe table table-striped table-borderless table-hover"
          data-testid="cpfcontributions"
        >
          <thead>
            <tr>
              <th>
                For Month
              </th>
              <th>
                Paid On
              </th>
              <th
                class="text-end"
              >
                Amount (S$)
              </th>
              <th>
                Employer
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Aug 2021
              </td>
              <td>
                25 Aug 2021
              </td>
              <td
                class="text-end"
              >
                1425
              </td>
              <td>
                CRYSTAL HORSE INVEST PTE LTD
              </td>
            </tr>
            <tr>
              <td>
                Sep 2021
              </td>
              <td>
                25 Sep 2021
              </td>
              <td
                class="text-end"
              >
                1425
              </td>
              <td>
                CRYSTAL HORSE INVEST PTE LTD
              </td>
            </tr>
          </tbody>
        </table>
        <div>
          <button
            class="button text-uppercase btn btn-primary"
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h5
        class="my-4"
      >
        Notice of Assessment (History)
      </h5>
      <div
        data-testid="noas"
      >
        <div
          class="mb-3 row"
        >
          <div
            class="text-start fw-bolder col-md-6"
          >
            Year of Assessment
          </div>
          <div
            class="text-end col-md-3"
          >
            2021
          </div>
          <div
            class="text-end col-md-3"
          >
            2020
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start col-md-6"
          >
            Employment
          </div>
          <div
            class="text-end col-md-3"
          >
            64400
          </div>
          <div
            class="text-end col-md-3"
          >
            36112.8
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start col-md-6"
          >
            Trade
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start col-md-6"
          >
            Interest
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start col-md-6"
          >
            Rent
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
          <div
            class="text-end col-md-3"
          >
            0
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start fw-bolder col-md-6"
          >
            Total income
          </div>
          <div
            class="text-end col-md-3"
          >
            64400
          </div>
          <div
            class="text-end col-md-3"
          >
            53700
          </div>
        </div>
        <div
          class="mb-3 row"
        >
          <div
            class="text-start fw-bolder col-md-6"
          >
            Tax Clearance
          </div>
          <div
            class="text-end col-md-3"
          >
            N
          </div>
          <div
            class="text-end col-md-3"
          >
            N
          </div>
        </div>
      </div>
      <h5
        class="my-4"
      >
        Other income Information
      </h5>
      <label
        class="text-black-50 mb-1 form-label"
        for="ownerprivate"
      >
        Ownership of Private Residential Property
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="ownerprivate"
          id="ownerprivate"
          readonly=""
          type="text"
          value="No"
        />
      </div>
      <h5
        class="my-4"
      >
        CPF Account Balance
      </h5>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          Ordinary Account (OA)(S$)
        </span>
        <input
          aria-describedby="basic-addon1"
          class="text-end form-control"
          data-testid="cpfbalances_oa"
          readonly=""
          type="text"
          value="58839.75"
        />
      </div>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon2"
        >
          Special Account (SA)(S$)
        </span>
        <input
          aria-describedby="basic-addon2"
          class="text-end form-control"
          data-testid="cpfbalances_sa"
          readonly=""
          type="text"
          value="15349.5"
        />
      </div>
      <div
        class="mb-3 input-group"
      >
        <span
          class="input-group-text"
          id="basic-addon3"
        >
          Medisave Account (MA)(S$)
        </span>
        <input
          aria-describedby="basic-addon3"
          class="text-end form-control"
          data-testid="cpfbalances_ma"
          readonly=""
          type="text"
          value="20466"
        />
      </div>
      <h5
        class="my-4"
      >
        CPF Contributions History
      </h5>
      <table
        class="cpf_talbe table table-striped table-borderless table-hover"
        data-testid="cpfcontributions"
      >
        <thead>
          <tr>
            <th>
              For Month
            </th>
            <th>
              Paid On
            </th>
            <th
              class="text-end"
            >
              Amount (S$)
            </th>
            <th>
              Employer
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Aug 2021
            </td>
            <td>
              25 Aug 2021
            </td>
            <td
              class="text-end"
            >
              1425
            </td>
            <td>
              CRYSTAL HORSE INVEST PTE LTD
            </td>
          </tr>
          <tr>
            <td>
              Sep 2021
            </td>
            <td>
              25 Sep 2021
            </td>
            <td
              class="text-end"
            >
              1425
            </td>
            <td>
              CRYSTAL HORSE INVEST PTE LTD
            </td>
          </tr>
        </tbody>
      </table>
      <div>
        <button
          class="button text-uppercase btn btn-primary"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PersonalInfo renders PersonalInfo component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h5
          class="my-4"
        >
          Personal Info
        </h5>
        <label
          class="text-black-50 mb-1 form-label"
          for="uinfin"
        >
          NRIC/FIN
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="uinfin"
            id="uinfin"
            readonly=""
            type="text"
            value="S9812381D"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="name"
        >
          Principal Name
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="name"
            id="name"
            readonly=""
            type="text"
            value="TAN XIAO HUI"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="sex"
        >
          Sex
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="sex"
            id="sex"
            readonly=""
            type="text"
            value="FEMALE"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="dob"
        >
          Date of Birth
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="dob"
            id="dob"
            readonly=""
            type="text"
            value="1998-06-06"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="birthcountry"
        >
          Country of Birth
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="birthcountry"
            id="birthcountry"
            readonly=""
            type="text"
            value="SINGAPORE"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="residentialstatus"
        >
          Residential Status
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="residentialstatus"
            id="residentialstatus"
            readonly=""
            type="text"
            value="CITIZEN"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="nationality"
        >
          Nationality
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="nationality"
            id="nationality"
            readonly=""
            type="text"
            value="SINGAPORE CITIZEN"
          />
        </div>
        <label
          class="text-black-50 mb-1 form-label"
          for="race"
        >
          Race
        </label>
        <div
          class="mb-3 input-group"
        >
          <input
            class="form-control-plaintext"
            data-testid="race"
            id="race"
            readonly=""
            type="text"
            value="CHINESE"
          />
        </div>
        <div>
          <button
            class="button text-uppercase btn btn-primary"
            type="button"
          >
            Continue
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h5
        class="my-4"
      >
        Personal Info
      </h5>
      <label
        class="text-black-50 mb-1 form-label"
        for="uinfin"
      >
        NRIC/FIN
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="uinfin"
          id="uinfin"
          readonly=""
          type="text"
          value="S9812381D"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="name"
      >
        Principal Name
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="name"
          id="name"
          readonly=""
          type="text"
          value="TAN XIAO HUI"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="sex"
      >
        Sex
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="sex"
          id="sex"
          readonly=""
          type="text"
          value="FEMALE"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="dob"
      >
        Date of Birth
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="dob"
          id="dob"
          readonly=""
          type="text"
          value="1998-06-06"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="birthcountry"
      >
        Country of Birth
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="birthcountry"
          id="birthcountry"
          readonly=""
          type="text"
          value="SINGAPORE"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="residentialstatus"
      >
        Residential Status
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="residentialstatus"
          id="residentialstatus"
          readonly=""
          type="text"
          value="CITIZEN"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="nationality"
      >
        Nationality
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="nationality"
          id="nationality"
          readonly=""
          type="text"
          value="SINGAPORE CITIZEN"
        />
      </div>
      <label
        class="text-black-50 mb-1 form-label"
        for="race"
      >
        Race
      </label>
      <div
        class="mb-3 input-group"
      >
        <input
          class="form-control-plaintext"
          data-testid="race"
          id="race"
          readonly=""
          type="text"
          value="CHINESE"
        />
      </div>
      <div>
        <button
          class="button text-uppercase btn btn-primary"
          type="button"
        >
          Continue
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
